src/test/java/com/techsophy/tsf/account/service/UserFormDataServiceTest.java

```diff
@@ -113,6 +115,33 @@ void saveUserTest() throws IOException
         userFormDataService.saveUserFormData(userFormDataSchema1);
         verify(mockUserFormDataDefinitionRepository,times(2)).save(any());
     }
+    @Test
+    void saveUserTestWithName() throws IOException
+    {
+        ObjectMapper objectMapper=new ObjectMapper();
+        InputStream stream = new ClassPathResource(USER_DATA).getInputStream();
+        String userData= new String(stream.readAllBytes());
+        UserFormDataSchema userFormDataSchema= objectMapper.readValue(userData,UserFormDataSchema.class);
+        UserFormDataSchema userFormDataSchema1 = new UserFormDataSchema(map,"1","1");
+        UserFormDataDefinition userFormDataDefinition = new UserFormDataDefinition(BigInteger.valueOf(1),map,BigInteger.valueOf(1),1);
+        UserFormDataDefinition userFormDataDefinition1 = new UserFormDataDefinition(null,map,null,1);
+        UserData userSchema=objectMapper.convertValue(userFormDataSchema.getUserData(), UserData.class);
+        UserDefinition userDefinition=new UserDefinition();
+        userDefinition.setFirstName("user");
+        userDefinition.setCreatedById(BigInteger.valueOf(234234234));
+        userDefinition.setId(BigInteger.valueOf(345345));
+        when(mockUserServiceImpl.getCurrentlyLoggedInUserId()).thenReturn(list);
+        when(mockUserServiceImpl.saveUser(any())).thenReturn(userDefinition.withId(BigInteger.valueOf(1234)));
+        when(mockObjectMapper.convertValue(any(),eq(UserFormDataDefinition.class))).thenReturn(userFormDataDefinition).thenReturn(userFormDataDefinition1);
+        when(mockObjectMapper.convertValue(any(),eq(UserFormDataSchema.class))).thenReturn(userFormDataSchema1);
+        when(mockObjectMapper.convertValue(any(),eq(UserData.class))).thenReturn(userSchema);
+        when(mockObjectMapper.convertValue(userFormDataDefinition,UserFormDataSchema.class)).thenReturn(userFormDataSchema1);
+        when(mockUserFormDataDefinitionRepository.save(any())).thenReturn(userFormDataDefinition);
+        System.out.println("username="+map.get("userName"));
+        UserFormDataSchema response=userFormDataService.saveUserFormData(userFormDataSchema);
+        Assertions.assertNotEquals(response.getUserData().get("userName"),"Nandini");
```

Comment:
sorry, not able to see any input for username Nandini, you need to provide the name and check that its stored(repository) in lower case.

Reply:
not alone verify in response